edits to examples text and a few tweaks (#29)
boettiger-lab/rl-intro: manuscript/manuscript.Rmd

The manager seeks to optimize the net present value (discounted cumulative catch) of a fishery, observing the stock size each year and setting an appropriate harvest quota in response.
In the classical approach, the best model of the fish population dynamics must first be estimated from data, potentially with posterior distributions over parameter estimates reflecting any uncertainty.
From this model, the optimal harvest policy -- that is, the function which returns the optimal quota for each possible observed stock size -- can be determined either by analytic [@Reed1979] or numerical [@Marescot2013] methods, depending on the complexity of the model.
In constrast, a model-free deep RL algorithm makes no assumption about the precise functional form or parameter values underlying the dynamics -- it is in principle agnostic to the details of the simulation.

We illustrate the deep RL approach using the algorithm known as Twin Delayed Deep Deterministic Policy Gradient or more simply, TD3 [@TD3].
TD3 is an extension of Deep Deterministic Policy Gradient [DDPG, @DDPG], which itself extended the original Deep Q-Network [DQN, @DQN] algorithm to continuous action space.

> A step-by-step walk-through for training both agents on this environment is provided in the Appendix.

Reviewer: ```suggestion
A step-by-step walk-through for training agents on this environment is provided in the Appendix.
```

Nevertheless, comparing the mean state over replicate simulations reveals some differences in the RL strategy, wherein the stock is maintained at a slightly higher-than-optimal biomass.
Because our state space and action space are sufficiently low-dimensional in this example, we are also able to visualize the policy function directly, and compare to the optimal policy [Fig 2].
This confirms that quotas tend to be slightly lower than optimal, most notably at larger stock sizes.
These features highlight a common challenge in the design and training of RL algorithms.
RL cares only about improving the realized cumulative reward, and may sometimes achieve this in unexpected ways.
Because these simulations rarely reach stock sizes at or above carrying capacity, these larger stock sizes show a greater deviation from the optimal policy than we observe at more frequently visited lower stock sizes.
Training these agents in a variety of alternative contexts can improve there ability to generalize to other scenarios.

How would an RL agent be applied to empirical data?
In principle, this is straight forward.

> Once our agent has been trained on appropriate simulation environments, it is able to propose a quota given an observation of the stock after appropriate transformation of variables.

Reviewer: ```suggestion
After we train an agent on a simulation environment that approximates the fishery of interest, we can query the policy of the agent to find a quota for the observed stock.    
```

Nevertheless, comparing the mean state over replicate simulations reveals some differences in the RL strategy, wherein the stock is maintained at a slightly higher-than-optimal biomass.
Because our state space and action space are sufficiently low-dimensional in this example, we are also able to visualize the policy function directly, and compare to the optimal policy [Fig 2].
This confirms that quotas tend to be slightly lower than optimal, most notably at larger stock sizes.
These features highlight a common challenge in the design and training of RL algorithms.
RL cares only about improving the realized cumulative reward, and may sometimes achieve this in unexpected ways.
Because these simulations rarely reach stock sizes at or above carrying capacity, these larger stock sizes show a greater deviation from the optimal policy than we observe at more frequently visited lower stock sizes.
Training these agents in a variety of alternative contexts can improve there ability to generalize to other scenarios.

How would an RL agent be applied to empirical data?
In principle, this is straight forward.

> Once our agent has been trained on appropriate simulation environments, it is able to propose a quota given an observation of the stock after appropriate transformation of variables.

Reviewer: Did we talk about state space transformation above? I feel like we could get away with leaving the transformation bit out and let the motivated see it in appendix B.

To illustrate this, we examine the quota that would be recommended by our newly trained RL agent, above, against historical harvest levels of Argentine hake based on stock assessments from 1986 - 2014 [@ram, see Appendix D].

> Hake stocks showed a marked decline throughout this period, while harvests deceased only in proportion [Fig 3].

Reviewer: ```suggestion
Hake stocks showed a marked decline throughout this period, while harvests decreased only in proportion [Fig 3].
```

```

This approach is not as different from conventional strategies at it may seem.
In a conventional approach, ecological models are first estimated from empirical data, (stock assessments in the fisheries case).

> Quotas can then be set based directly on these model estimates, or by comparing alternative candidate "harvest control rules" (the policy function) against model-based simulations of stock dynamics.

Reviewer: ```suggestion
Quotas can then be set based directly on these model estimates, or by comparing alternative candidate "harvest control rules" (policies) against model-based simulations of stock dynamics.
```

This latter approach, known in fisheries as Management Strategy Evaluation [MSE; @Punt2016] is already very closely analogous to the RL process.

> Instead of researchers evaluating a handful of control rules, the RL agent takes on the task of proposing and evaluating the millions of possible control rules represented by its neural network. 

Reviewer: ```suggestion
Instead of researchers evaluating a handful of control rules, the RL agent proposes and evaluates a plethora of possible control rules autonomously. 
```
It's hard for me to fit in neural networks in this sentence in a way that gibes with value-based methods for which the NN does not represent the policy. Besides, I think mentioning NNs might be distracting here to people that aren't familiar with NNs. The main point, I think you are trying to convey, is that RL tries a lot of policies/control rules and sees what works best.
